### PR TITLE
Add coverage in CI

### LINKED ISSUE #826: Repository test coverage
Need to connect our main `gear` repo with some kind of test-coverage-viewer.

### PR DESCRIPTION
**Pull request by ark0f**
Resolves #826 

Note: WASM cannot be build with `-Cinstrument-coverage` flag because the target does not have rustc's `profiler_builtins` crate

### CODE REVIEW

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -283,3 +283,125 @@ jobs:
...
+        run: |
+          curl -L https://github.com/Kogia-sima/rust-covfix/releases/download/v0.2.2/rust-covfix-linux-x86_64.tar.xz | tar Jxf -
+          mv rust-covfix-linux-x86_64/rust-covfix ./
+
+      - name: "Cache: Unpack"
+        if: github.event.pull_request.merged == false
+        continue-on-error: true
+        run: |
+          cp /root/cache/gear-test* /tmp/
+          tar -xf /tmp/gear-test_cache.tar -C /
```
Comment (ukint-vs) [inline .github/workflows/CI.yaml:341]: also add `build_*.tar` lines

File: .github/workflows/CI.yaml [hunk 2]
```diff
@@ -283,3 +283,125 @@ jobs:
...
+        
+      - name: "Coverage: Fix report"
+        run: ./rust-covfix -o lcov.info lcov.info
+        
+      - name: "Coverage: Publish"
+        uses: codecov/codecov-action@v1
+        with:
+          file: ./lcov.info
+
+      - name: "Cache: Pack"
```
Comment (ukint-vs) [inline .github/workflows/CI.yaml:400]: I think packing cache from code coverage is not needed

**Review by ukint-vs — APPROVED** [review]

File: .github/workflows/CI.yaml (hunk 2 above)
Comment (ark0f) [reply .github/workflows/CI.yaml:400]: Do we need unpacking then?

Comment (ukint-vs) [reply .github/workflows/CI.yaml:400]: yes, as it will speed up the build time

**Review by ukint-vs — APPROVED** [review]

**Review by NikVolf — APPROVED** [review]
> 65.04% 

pathetic though

**Review by ukint-vs — APPROVED** [review]
great!

**Pull request merged**